error: to use a constant of type `TypeId` in a pattern, `TypeId` must be annotated with `#[derive(PartialEq, Eq)]`
error: could not compile `value-bag` (lib) due to 3 previous errors

error: failed to run custom build command for `ring v0.16.20`

---

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,13 +34,15 @@ serde = "1.0.124"
 serde_derive = "1.0.116"
 cidr-utils = "0.5.1"
 itertools = "0.9.0"
-trust-dns-resolver = { version = "0.23.2", features = ["dns-over-rustls"] }
+trust-dns-resolver = { version = "=0.23.0", features = ["dns-over-rustls"] }
 anyhow = "1.0.40"
 subprocess = "0.2.6"
 text_placeholder = { version = "0.5", features = ["struct_context"] }
 
 [dev-dependencies]
 wait-timeout = "0.2"
+sct = "0.7.1"
+value-bag = "1.4.1"
 
 [package.metadata.deb]
 depends = "$auto, nmap"
@@ -54,4 +56,8 @@ strip = true
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 [[bin]]
 name = "rustscan"
-path = "src/main.rs"
\ No newline at end of file
+path = "src/main.rs"
+
+[patch.crates-io]
+# mips/mipsel: error: failed to run custom build command for `ring v0.16.20`
+trust-dns-resolver = { path = "../trust-dns-0.23.0/crates/resolver" }
